### ISSUE TITLE
Anthrax Bomb and Toxin Tractor Contamination can be cleared by enemy Poison and Radiation Clouds

### ISSUE THREAD

**Issue by commy2** [TheSuperHackers/GeneralsGamePatch#55]
When the enemy drops an Anthrax bomb, force fire the ground near with a Toxin Shell Scorpion or Nuke Battlemaster to delete it.

### Proposal 1
Stacked poison no longer disappears.

### Proposal 2
Stacked poison only disappears for ally attacks.

### Proposal 3
Anthrax bomb spawns multiple puddles and requires more shots at different locations to be cleared.

### Proposal 4
If exploit is removed entirely, then reduce Anthrax Bomb radius to mitigate impact on gameplay.

### Proposal 5
If exploit is removed entirely, then reduce Anthrax Bomb poison duration to mitigate impact on gameplay.

### Proposal 6
If exploit is removed entirely, then reduce Anthrax Bomb poison damage to mitigate impact on gameplay.

### Proposal 7
Allow China to remove toxins with fire if we keep this mechanic to keep it consistent across factions.

### Proposal 8
Anthrax Bomb puddle requires more damage to die. (Done)
* #2023

**Comment by commy2** [comment]
Why is this even a thing?

This is done to prevent poison clouds from stacking, but this also makes it so the player can just delete enemy large poison fields by creating small ones in place. 

What is my suggestion?

Ideally poison fields could only clear up allied poison fields. That way stacking is prevented for the most part without being able to overwrite enemy poison attacks altogether. Implementation can be seen here: https://github.com/commy2/zerohour/commit/c5ba420cd7a7d0b0a2db1cf157ccc0510cabedf7

**Comment by xezon** [comment]
I highly prefer a clean fix for this, then adjust balance if necessary. The current way the anthrax can be removed with 1 shot is not satisfying behavior and does not make the gameplay better.

**Comment by xezon** [comment]
Added proposals to opening post. I still think clean fix is best and then mitigate the inherent buff of Anthrax Bomb by tweaking it. Meaning Proposal 1+4 or 1+5 for example.

Proposal 2 I think is confusing.

Proposal 3 is better than status quo, but not really a good solution. Radiation also does not disappear on force fire. So why should Toxins?

**Comment by commy2** [comment]
The issue with proposal 1 is, that it means that poison/radiation clouds can now stack, which is a damage buff to Nuke Battle Masters/Overlords and Toxin Scorpions/Marauders. It may also affect Toxin Trucks/Rebels, but I don't know if and how bad that is.

Deleting allied poison clouds is reasonable, not deleting any of them at all affects a lot of things that are not the Anthrax Bomb.

**Comment by xezon** [comment]
Do we know if radiation puddles stack? And if not, why not?

**Comment by commy2** [comment]
Nuke General radiation puddle has 100 health:

https://github.com/TheSuperHackers/GeneralsGamePatch/blob/main/Patch104pZH/GameFilesOriginalZH/Data/INI/Object/NukeGeneral.ini#L28-L29

When it is created, it fires the `HazardFieldCoreWeapon`.

https://github.com/TheSuperHackers/GeneralsGamePatch/blob/main/Patch104pZH/GameFilesOriginalZH/Data/INI/Object/NukeGeneral.ini#L41

That weapon does 200 damage with HAZARD_CLEANUP type.

https://github.com/TheSuperHackers/GeneralsGamePatch/blob/main/Patch104pZH/GameFilesOriginalZH/Data/INI/Weapon.ini#L3406-L3408

Thus, new fields delete old fields. This works similar / the same for all poison and radiation fields.

**Comment by xezon** [comment]
Ok. So behaviour is consistent then. I thought it was Toxin only.

**Comment by commy2** [comment]
Yeah, and that is why I came up with proposal 2. It keeps the intend and unit balance, but gets rid of the exploits. However, one would possibly need to compensate since it is a buff to xGLA in matchups vs Nuke, Tox and lategame vGLA.

**Comment by xezon** [comment]
Ok I understand. What happens if you use something other than ALLIES, such as SUICIDE ?

**Comment by commy2** [comment]
SELF - damage self
ALLIES - damage other units from the same player or a team mate in 2v2 etc.
ENEMIES - damage enemy units
NEUTRALS - damage civilian cars, buildings etc.
NOT_SIMILAR - do NOT damage __allied__ units of the same type, for projectiles, do not damage __allied__ projectiles of the same type (e.g. ScudMissile cannot blow up other ScudMissile).
SUICIDE - no idea, same as SELF?

I propose to remove NEUTRALS and ENEMIES from `HazardFieldCoreWeapon` and keeping only ALLIES. Obviously do not add SELF or SUICIDE, otherwise a poison field would clear itself.

**Comment by MTKing4** [comment]
Ok so with proposal 2 puddles will still not stack but enemies can't clear it, sounds like a reasonable change, but yeah i do feel like a proper toxin cleaning unit is necessary for all factions

I suggest:
1- Ambulance for USA (obviously)
2- Toxin Tractor GLA ( A new Clean toxin ability, not the same sprinkler)
3- for China the options are:
    (a) Troop Crawler (add same ability, but would need a turret)
    (b) Outpost (Same ability, might need a turret)
    (c) Dragon tank (Same ability or firewall)

**Comment by commy2** [comment]
Well, in the CCG China campaign mission 4, the story makes a point of how you have to use a Mig strike to burn chemicals from a GLA toxin factory, otherwise they'd spill into and pollute a nearby river. I always found it weird how despite this, flame weapons cannot actually clear poison at all.

Granted, it wouldn't really make sense for radiation puddles.

**Comment by xezon** [comment]
Can Dragon Tank clear Anthrax Bomb in original game? I think Troopcrawler and Outpost are not fit for clearing toxins, because they die quickly to toxins. And changing that would change dynamics, especially for China Infantry.

**Comment by MTKing4** [comment]
I updated the comment

**Comment by commy2** [comment]
> Can Dragon Tank clear Anthrax Bomb in original game?

No, but without having tried, one could just make poison puddles vulnerable to FLAME type damage, and then they should be clearable by force-fire.

**Comment by xezon** [comment]
But then we are back at square one with clearing toxins by force firing.

**Comment by MTKing4** [comment]
> think Troopcrawler and Outpost are not fit for clearing toxins, because they die quickly to toxins

Ambulances die quickly to toxins as well, unless you start clearing.

**Comment by Jundiyy** [comment]
Whilst I'd like a fix, feel it would be quite a big change, other than proposal 3. Whilst it is still a bigger change, it still works the way it is now.

**Comment by commy2** [comment]
> But then we are back at square one with clearing toxins by force firing.

Kinda, yeah. Ambulance can already clear the Tox Bomb in one shot though.

>Ambulances die quickly to toxins as well, unless you start clearing.

Ambulance is immune to poison (and radiation). It does only take ~~a small amount of~~ damage by the initial detonation of the Anthrax Bomb, but that is nowhere near enough to kill it.

**Comment by xezon** [comment]
USA Ambulance OP.

**Comment by commy2** [comment]
> Whilst I'd like a fix, feel it would be quite a big change, other than proposal 3.

Prop 3 does not address the Toxin Truck though.

**Comment by MTKing4** [comment]
> > Can Dragon Tank clear Anthrax Bomb in original game?
> 
> No, but without having tried, one could just make poison puddles vulnerable to FLAME type damage, and then they should be clearable by force-fire.

In that case even migs can clear toxins, it couldn't be vulnerable only to flame walls?

**Comment by commy2** [comment]
> In that case even migs can clear toxins, it couldn't be vulnerable only to flame walls?

No, all FLAME weapons. Including the sparks from Inferno Cannons and MOABs.

**Comment by Jundiyy** [comment]
That is true, didn't think of that, although that at the most, would just disturb the Supply lines or a Barracks. Which isn't as bad as a Tox Bomb.

**Comment by xezon** [comment]
It would only be fair then that China would be able to exploit the bug with fire as well if we keep the bug... but ideally there would be no bug to begin with.

**Comment by Jundiyy** [comment]
True, would be best to keep it fair, but can't see any other 'natural' work around.

**Comment by xezon** [comment]
Added Proposal 7: Allow China to remove toxins with fire if we keep this mechanic to keep it consistent across factions.

Proposal 7 can combine with Proposal 3.

**Comment by commy2** [comment]
Tox Bomb is also pretty devastating in GLA mirrors. Since you can block multiple Arms Dealers with one strike.

Although I realize that GLA mirrors are fundamentally broken already.

**Comment by xezon** [comment]
The problem is that Anthrax Bomb just lasts so very long.

**Comment by MTKing4** [comment]
Dragon Tank seems like the Perfect Candidate

https://user-images.githubusercontent.com/83613147/180619596-2c6220ab-d610-4f65-ba04-2af009d2099c.mp4

**Comment by commy2** [comment]
Made a branch to mess around with FLAME weapons vs poison.

**Comment by ExiLe-ZH** [comment]
> Although I realize that GLA mirrors are fundamentally broken already.

Besides this bug (that changes alot), gla mirrors are played as they're supposed to be?

> Made a branch to mess around with FLAME weapons vs poison.

Would be cool if the toxins could spread the fire, same like burning trees. If possible?

**Comment by commy2** [comment]
> Besides this bug (that changes alot), gla mirrors are played as they're supposed to be?

BM spam camping.

> Would be cool if the toxins could spread the fire, same like burning trees. If possible?

Probably possible, but would take more time than I want to commit right now for a test branch.  Later perhaps.

>  2- Toxin Tractor GLA ( A new Clean toxin ability, not the same sprinkler)

Issue with this is Demo General. Demo Toxin Truck already has 3 weapons: poison stream, poison sprinkler, and suicide.  You can only have three weapons by engine limitations.  Should totally be possible for the other three tractors though.

**Comment by MTKing4** [comment]
Then i guess no change needed, we're going with the Forcefire decision anyway so this can remain as it is.

**Comment by xezon** [comment]
Keeping the force fire behavior as is to remove poison and radiation looks reasonable to me, because it appears to be a deliberate game feature to avoid damage stacking as outlined by @commy2 .

Allowing China to clean poison just like GLA and USA can, also reads logical to me. This puts everyone at an even playing field when getting shot at with poison. However, can USA and GLA clean Radiation? If not, should they be able to?

And then last question would be about Anthrax Bomb. To me it looks broken that 5 Star Genpower can be deactivated with 1 shot at its core. So is splitting it to 4, 8 or 12 puddles an option? If not, why?

**Comment by commy2** [comment]
> can USA clean Radiation

Yes, the Ambulance can do that as well. Same as poison

>can GLA clean Radiation

In 1.04 they can by force-firing the ground with a Tracor, or if vGLA or Tox, with a Scorpion or Marauder.

> So is splitting it to 4, 8 or 12 puddles an option? If now, why?

This is Jundiyy's favoured solution. However, to me it would just mean that you clean up the bomb in 1.5 seconds with mutliple clicks, instead of in 1 second with a single click. I don't see how that improves anything.

**Comment by ExiLe-ZH** [comment]
multiple puddles is my preferred option aswell, it definitely would take more time and attention than 1.5 seconds.

**Comment by commy2** [comment]
Okay, maybe 3.

**Comment by ExiLe-ZH** [comment]
> Okay, maybe 3.

Tox bomb area is quite big, if it's like 12 puddles and your unit needs to move around it's atleast 1 second per puddle.

**Comment by commy2** [comment]
It's also difficult to arrange circular puddles in a bigger circle without creating any overlaps that would change how much damage units can take. If no overlap, then some units would not take damage at all even though they are inside the area. If full overlap, some units would take twice (sometimes thrice) as much damage. If something in between, both could happen at the same time. It just seems like a very convulted thing to arrange reasonably, if that is possible at all.

**Comment by ExiLe-ZH** [comment]
> It's also difficult to arrange circular puddles in a bigger circle without creating any overlaps that would change how much damage units can take. If no overlap, then some units would not take damage at all even though they are inside the area. If full overlap, some units would take twice (sometimes thrice) as much damage. If something in between, both could happen at the same time. It just seems like a very convulted thing to arrange reasonably, if that is possible at all.

I believe some popular mod already has this implemented

**Comment by commy2** [comment]
I have heard that many times at this point, not just regarding this. Turned out either that wasn't the case at all (ROTR and PDL drone dummies) or the implementation had severe problems.

**Comment by xezon** [comment]
What commy2 writes is factually true. You can't geometrically arrange circles without overlaps or holes. This only works for shapes that have aligning borders, such as rectangles. A solution would therefore require circle overlaps and something that prevents damage stacking where they do overlap. If such functionality does not exist, then you can only have a dirty fix, which will result in Anthrax Bomb causing different damages at seemingly random positions.

**Comment by Jundiyy** [comment]
I am not pushing for this just because I proposed it.
Contra Mod has done this, I might be able to get a video for you.

Regarding clearing up, it does take more than a few seconds since they have 7 or 8 puddles.

**Comment by Jundiyy** [comment]
> What commy2 writes is factually true. You can't geometrically arrange circles without overlaps or holes. This only works for shapes that have aligning borders, such as rectangles. A solution would therefore require circle overlaps and something that prevents damage stacking where they do overlap. If such functionality does not exist, then you can only have a dirty fix, which will result in Anthrax Bomb causing different damages at seemingly random positions.

That is true, hanfield did say long ago that you could get it quite close to original. Of course, wouldn't be 100%.

**Comment by xezon** [comment]
For a test you could place a lot of workers standing close together on a large area and then drop the Anthrax Bomb on it. See if any worker would survive. If yes, then Anthrax has bigger holes.

**Comment by Jundiyy** [comment]
https://www.youtube.com/watch?v=gzd3FJnSSUU

**Comment by MTKing4** [comment]
I Have Uploaded Some Video Tests Regarding Proposal 7: Allow Flames To Clear Toxins

Check them out [Here](https://github.com/TheSuperHackers/GeneralsGamePatch/pull/749#issuecomment-1193368266)

**Comment by xezon** [comment]
Very thorough testing. Impressive.

**Comment by ExiLe-ZH** [comment]
Here my opinion about the whole thing:

1. One shot removal with anything is just silly, should be puddles in any case.

2. Exploit should be removed all together and puddles should be implemented to nerf the ambulance (maybe another method could achieve this too?)

3. To reduce impact on gameplay I would choose for proposal 5 and reduce the duration, cause it's really long.

4. This is probably not possible, but another way to prevent stacking could be implementing a stacking order? Where big radiation/toxin fields are prioritized to stay over small ones?

**Comment by commy2** [comment]
> 4 

Don't see how, sadly.

Another thought: All poison puddles have the same health points. One could increase the health of the Anthrax Bomb puddle, and then it would (perhaps) take longer to clear up by other puddles / the Ambulance / Fire.

**Comment by ZekeDlyoung** [comment]
Crazy idea for having multiple puddles but no overlap damage:
Turn the Large Anthrax field into an "Angry Mob Nexus" with the smaller fields being its members to create several smaller fields, but the smaller fields themselves don't have the field weapon only the Nexus does. That way, Ambos will have to clear all the smaller fields in order to kill the "Large Field mob", but the damage will still remain the same as only the nexus is actually doing the damage.

**Comment by MTKing4** [comment]
That's Genius!

**Comment by ZekeDlyoung** [comment]
Conversely, we can just give the large anthrax field a lot more health, so it takes longer to destroy 🙃

**Comment by xezon** [comment]
Damn, this would work? Angry Anthrax Puddles, lol

**Comment by xezon** [comment]
Zeke:
> 1. Take an angry mob nexus, replace all if its members with a copy of anthrax field large (or whatever the scud launcher uses, iirc there is no medium) with the FireWeaponUpdate removed, so they don't do damage.

> 2. Take the FireWeaponUpdate from the current anthrax bomb field and add it to the angry mob nexus, that's where the actual damage will come from.

> Potential problems with this setup is not being able to make the smaller fields spawn at the correct places, since iirc the angry mob starts all bunched up, then the members slowly move out as the mob moves.

**Comment by xezon** [comment]
commy2:
> Toxin Truck doesn't even work that well against Anthrax Bomb. I see it misused all the time. Force firing the ground doesn't work, the Tractor only "decontaminates" if it is blown up or uses the Sprinkler.

## Contamination Times

| Object                   | Contamination Time (s) | Range |
|--------------------------|------------------------|-------|
| Default Puddle           | 10                     |
| Toxin Tractor Sprinkler  | 180                    |
| SCUD Storm               | 30                     |
| SCUD Launcher Anthrax    | 30                     |
| Bio Bomb Truck           | 30                     |
| Demo Trap                | 30                     |
| Toxin Bomb               | 60                     |

**Comment by kABUSE1** [comment]
What is the reason why you don't just allow stacking, as shown on these screenshots?
https://www.gamereplays.org/community/index.php?s=&showtopic=573000&view=findpost&p=6808161

I don't really see how this would negatively impact gameplay. It could make the tox scud launchers better, as the damage of the tox puddles stacks. Also it might turn tox trucks into a decent weapon, that is not only used for clearing buildings (or removing anthrax bomb puddles)

**Comment by xezon** [comment]
The prevention of toxin stack is an intentional feature with explicit documentation from EA developers. It does not explain why, but we can assume that the intention was to not end up with huge amounts of stacked damages. It is not clear if stack prevention is ultimately desired, therefore this would have to be tested carefully.

First step would be to identify common weapons that would stack over short to long durations. Would a toxin puddle of a weapon stack with another puddle of the same weapon? How quickly can we expect units to die on stacked puddles, such as puddles from Toxin Rebel, Toxin Scorpion, Toxin Tractor and Anthrax Bomb? Is the stacked damage too much?

Perhaps a test map would be good to see a representative selection of toxin stack samples to anticipate the consequences more accurately.

If I am not mistaken, stacking is also a topic for radiation puddles.

**Comment by kABUSE1** [comment]
I see.

Something to keep in mind 

MTKing:
> That one is about the death of the toxin truck, when it dies it spawns a small toxin puddle that that overwrites the contamination puddle, different issue
> Way to avoid that was to basically not let the toxin truck die on top of the contamination puddle i.e activate contamination then move the truck away
> Many players faced that issue because they just leave the truck contaminating until it dies

So even if you redesigned tox puddle behaviour so that only ally tox puddles are able to remove each other, a tox truck dying while using the area "Contamination" ability would still cancel the large contamination puddle out with the small death tox puddle.

So simply making this "friendly fire only" doesn't seem to be the ultimate fix.

**Comment by kABUSE1** [comment]
> Perhaps a test map would be good to see a representative selection of toxin stack samples to anticipate the consequences more accurately.

I can build that test map if somebody can tell me how to make this stuff stackable via .ini.

I assume would need to be done for each individual variant of radiation and tox puddle in the game.

**Comment by xezon** [comment]
Test map would not modify logic, so it can be tested against any setups.

The weapon responsible for toxin removal is `Weapon HazardFieldCoreWeapon`.

**Comment by xezon** [comment]
Proposal 8 added.

**Comment by xezon** [comment]
So what do we do here? :D

**Comment by ExiLe-ZH** [comment]
For me proposal 3 + 7

**Comment by xezon** [comment]
I do not know how to implement nor if it is possible to implement Proposal 3 in INI.

Before going either direction, perhaps we can gather a list of Advantages and Disadvantages of Toxin and Radiation stacking to get an overview of what speaks for and against it?

Then it will be easier to make a judgement I think.

**Comment by MTKing4** [comment]
Proposal 7 & 8 for me

I think it's both doable and good middle ground both in providing counterplay for armies that suffer greatly from anthrax bomb and in mitigating that counterplay efficiency so that anthrax is not completely useless against certain armies

It also is the Smallest change possible and not very noticeable so that it will be easily adopted by the playerbase

**Comment by commy2** [comment]
2 & 7 and make it so Toxin Tractor does not spawn a poison puddle when killed while spraying.

Clearing stacked allied puddles likely exists to cap damage of fast firing poison weapons. Clearing enemy puddles is an oversight and exploit.

The advantage of keeping stack deletion is that 99% of situations remain unchanged from 1.04. Notable exception is probably Nuke vs Nuke Battle Masters suiciding into each other, which, if clearing enemy puddles is removed, makes intermingling puddles from both players stronger. Aside from that, stacking enemy poison clouds almost never happens unless the exploit is used. Closest thing I can come up with is Tox Scorps vs Tox Scorps, but they rarely are on top of each other like Nuke BMs.

**Comment by xezon** [comment]
Proposal 8 seems to be good regardless of what we do. I think it is always preferred that a larger puddle survives longer than a small one. Perhaps we can scale it so that small puddles have the least health, and larger one have more health. Then larger puddles will kill smaller ones.

**Comment by commy2** [comment]
HazardFieldCoreWeapon is only fired once the puddle is created, so making big puddles survive means that small puddles could be on top of larger ones without them disappearing themselves.

**Comment by xezon** [comment]
My understanding is that we can add firing interval here and redesign the damage to health distribution among puddles. Right now it is one shot. 200 Damage vs 100 to 150 Health of puddles.

```
;This weapon is used at the core of every damage field to kill other fields to prevent stacking
Weapon HazardFieldCoreWeapon
  PrimaryDamage = 200.0
  PrimaryDamageRadius = 10.0
  AttackRange = 15.0
  DamageType = HAZARD_CLEANUP

  RadiusDamageAffects = ALLIES ENEMIES NEUTRALS

  ; This clip pattern insures that the new field kills the old fields
  DelayBetweenShots = 5
  ClipSize = 1
  ClipReloadTime = 100000
End
```

**Comment by commy2** [comment]
Maybe, but you're likely going to create a mess where "weakened" puddles exist and old puddles stay fighting new puddles. It's something I would avoid if possible.

**Comment by xezon** [comment]
Why would that be a mess? Proposal 8 promises that large field would require more damage, so essentially more puddles would be needed to be placed on it.

**Comment by commy2** [comment]
firing intervals would create a mess.

**Comment by xezon** [comment]
I am not sure I follow. With this prototype it takes exactly 20 shots from Toxin Shells anywhere on the Anthrax Bomb to remove it.

* #2020

**Comment by xezon** [comment]
I could not observe any oddities. It works as expected.

Now the question is

1. How many Toxin Shell shots should remove an Anthrax Bomb?

2. How large should the removal radius of the Anthrax Bomb be? The maximum radius is 300, matching the Anthrax Bomb damage radius.

**Comment by ExiLe-ZH** [comment]
One issue there is with the whole removal mechanic is that's it's not a logical or documented ability/counter. I doubt it even was an intended mechanic to begin with. 

A newer player either has to learn it through passed knowledge or accidentally find out himself, when increasing HP the latter case will be a rare occurence.

**Comment by commy2** [comment]
> I doubt it even was an intended mechanic to begin with.

It's intended and documented in code. Using it to clear enemies is an oversight though.

**Comment by ExiLe-ZH** [comment]
> > I doubt it even was an intended mechanic to begin with.
> 
> Using it to clear enemies is an oversight though.

Has to be, one click to remove anthrax bomb is just ridicilous.

**Comment by xezon** [comment]
I noticed unpleasant issue with #2020. I set Anthrax Puddle of Scud Launcher to 50 health and 10 hazard damage, which will eliminate all puddles when 4 Scud Missiles hit the same location at the same time. So there are cases where all puddles would be able to kill each other to leave no behind. Not yet clear to me if that can be solved.

**Comment by xezon** [comment]
I can not think of a way to make #2020 work.

**Comment by MTKing4** [comment]
> 2 & 7 and make it so Toxin Tractor does not spawn a poison puddle when killed while spraying.

> Clearing stacked allied puddles likely exists to cap damage of fast firing poison weapons. Clearing enemy puddles is an oversight and exploit.

That's a good point, it will keep the original damage for toxin units so that nothing is changed, like when a toxin scorpion shoots on enemy, they won't stack their own toxins over it so no extra damage

Also one important factor, which i assume it is already the case in 1.04 is if a Scorpion is shooting the ground to clear anthrax, it's applying half of the damage on its own stacked puddles, so the optimal clearing method should be that you shoot at different locations to avoid stacking your own clouds, but i guess this isn't a new feature so it shouldn't matter too much. But i think preventing ally stacking is a good change regardless.

2 + 7 + 8 for me

**Comment by MTKing4** [comment]
> Proposal 8 seems to be good regardless of what we do. I think it is always preferred that a larger puddle survives longer than a small one. Perhaps we can scale it so that small puddles have the least health, and larger one have more health. Then larger puddles will kill smaller ones.

I think you need to be careful with that, we don't want larger puddles killing smaller ones before doing enough damage to the big one

I think all puddles should do the same amount of damage for simplicity and tweak their health, i think big puddle should have like 2 times the current health and the big puddles 4 times as much, so that it takes 2 shots to kill a small puddle and 4 shots to kill a big one, for starters you can go with that

**Comment by MTKing4** [comment]
> I noticed unpleasant issue with #2020. I set Anthrax Puddle of Scud Launcher to 50 health and 10 hazard damage, which will eliminate all puddles when 4 Scud Missiles hit the same location at the same time. So there are cases where all puddles would be able to kill each other to leave no behind. Not yet clear to me if that can be solved.

Preventing ally stacking would solve that pretty sure

**Comment by xezon** [comment]
The only way to fix #2020 I can see would be to guarantee that no 2 overlapping puddles spawn in the same logic frame. But we cannot do that.

**Comment by commy2** [comment]
> Also one important factor, which i assume it is already the case in 1.04 is if a Scorpion is shooting the ground to clear anthrax, it's applying half of the damage on its own stacked puddles, so the optimal clearing method should be that you shoot at different locations to avoid stacking your own clouds, but i guess this isn't a new feature so it shouldn't matter too much.

It shouldn't make any difference. The cleanup weapon does enough damage to one shot every other puddle. The only exception is easy AI firing on a hard AI puddle with 120 or more health (large puddles and Anthrax Bomb puddles only).

**Comment by xezon** [comment]
Boom, year 2023 brings pull #2023.
